### ISSUE TITLE
TLS handshake failed

### ISSUE THREAD

**Issue by imp1sh** [seniorgod/ifi-tidal-release#13]
Hi,

I think maybe the tidal app has been upgraded or the certificate has become invalid I don't know exactly, but the TLS handshake between the Android app and the ifi connect service won't work any more.

```
tidal_connect_application[1347]: [2022-08-20 10:22:23] [error] handle_transport_init received error: TLS handshake failed
```
This is what it looks like in tcpdump. 129.85 is my raspberry pi while 129.82 is my Android device.

![image](https://user-images.githubusercontent.com/3486156/185739309-686e6095-aa5e-4cc5-bcca-626de473aece.png)

**Comment by rasika-jay** [comment]
yep, can confirm
```
Aug 21 11:53:13 moode tidal_connect_application[366]: [2022-08-21 11:53:13] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:53:19 moode tidal_connect_application[366]: [2022-08-21 11:53:19] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:54:05 moode tidal_connect_application[366]: [2022-08-21 11:54:05] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:54:20 moode tidal_connect_application[366]: [2022-08-21 11:54:20] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:54:23 moode tidal_connect_application[366]: [2022-08-21 11:54:23] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:54:31 moode tidal_connect_application[366]: [2022-08-21 11:54:31] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:55:22 moode tidal_connect_application[366]: [2022-08-21 11:55:22] [error] handle_transport_init received error: TLS handshake failed
Aug 21 11:55:33 moode tidal_connect_application[366]: [2022-08-21 11:55:33] [error] handle_transport_init received error: TLS handshake failed
Aug 21 12:37:26 moode tidal_connect_application[366]: [2022-08-21 12:37:26] [error] handle_transport_init received error: TLS handshake failed
Aug 21 12:37:48 moode tidal_connect_application[366]: [2022-08-21 12:37:48] [error] handle_transport_init received error: TLS handshake failed
```

logs from service

**Comment by emerout** [comment]
Same issue here.

I have another android device whose tidal app has not been yet updated : tidal connect is still working as expected.

MacOs and iOs tidal applications are also still working.

It would be interesting to know if it's still working on a real Tidal Connect compatible device.

**Comment by imp1sh** [comment]
That really sounds that is an intentional shutdown of people like us by tidal. Would be really sad, because I will not invest in those overpriced Tidal Connect devices. The Raspberry worked not flawlessly but for the price extremely well for me together with gapless playback and Master quality.
Is there maybe some source we could search for an alternate certificate? The old one originates from this Ifi Zen stream, right? Any other ideas?

**Comment by lukeboga** [comment]
> Same issue here.
> 
> I have another android device whose tidal app has not been yet updated : tidal connect is still working as expected.
> 
> MacOs and iOs tidal applications are also still working.
> 
> It would be interesting to know if it's still working on a real Tidal Connect compatible device.

Yep - Still works on my Sonos (Play:1) with Tidal Connect but not the Pi (same error as above).....

**Comment by FelixL-FH** [comment]
I have the same issue. Any ideas where to get a new certificate?

**Comment by mcpattrsn** [comment]
It doesn't work with the phone app (android) but it works with the desktop app (windows).

**Comment by jbrepogmailcom** [comment]
I have read somewhere that it is still working with Android Tidal 2.49 ( https://forums.raspberrypi.com/viewtopic.php?t=297771&start=150 ). Can someone test it please?

As written here: https://community.volumio.org/t/lost-tidal-connect-after-update/57503/25

"We have contacted Tidal and the issue is in the Tidal app. They have fixed the issue and the fix will be released September the 21st." - apparently did not happen. "In the mean time, as a workaround, please reset the app’s cache when you encounter this issue."

**Comment by jbrepogmailcom** [comment]
I can confirm the android version 2.63 works. I have downloaded on PC and then downgraded with ```adb install -r -d tidal2_63.apk```
You cannot downgrade on mobile, you need to uninstall previous version and then you will loose all downloads. With ADB, it can downgrade and  use data from newer version

**Comment by wgorczyca** [comment]
https://github.com/TonyTromp/tidal-connect-docker/commit/8ab862b6df96b979269a4ea4419a317456230bee those certificates works with new android versions of tidal

**Comment by jbrepogmailcom** [comment]
My ifi stoppend working some weeks ago, I suspect that new certificates are needed...
